java-checks/src/main/java/org/sonar/java/checks/predicates/VisibleForTestingPredicate.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.java.checks.predicates;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.sonar.plugins.java.api.tree.AnnotationTree;
+import org.sonar.plugins.java.api.tree.ModifiersTree;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+/**
+ * This is a predicate that finds if given modifiers tree contains {@link VisibleForTesting} annotation.
+ *
+ * @author <a href="mailto:krzysztof.suszynski@coi.gov.pl">Krzysztof Suszynski</a>
+ * @since 07.06.16
+ */
+public class VisibleForTestingPredicate implements Predicate<ModifiersTree> {
+
+  public static final String GUAVA_FQCN = VisibleForTesting.class.getName();
```

Comment:
Yup. I've noticed that when pass it via String test validator can have `!UnknownSymbol` instead of correct `AnnotationTree` structure (while debugging the test). It must be due to yours custom lexer, which for some reason do not load `VisibleForTesting` class when encountered at test file like here: https://github.com/SonarSource/sonar-java/pull/848/files#diff-02c2ce19b6675f863c2cbecc70b95c23R15